xscen/utils.py

```diff
@@ -363,6 +368,9 @@ def clean_up(
     maybe_unstack_dict: dict
         Dictionary to pass to xscen.common.maybe_unstack function.
         The format should be: {'coords': path_to_coord_file, 'rechunk': {'time': -1 }, 'stack_drop_nans': True}.
+    common_attrs_only: dict, list
+        List of datasets, or path to NetCDF or Zarr files.
+        Keeps only the global attributes that are the same for all datasets and generates a new id.
```

Comment:
the generate a new id is missing in the code.

Reply:
I would put `generate_id` in the `common_attrs_only` code because I think the id will be deleted most of the time by `common_attrs_only` but ids are very useful when updating the catalog.

Reply:
Mmh, I'm getting a circular import crash since `catalog` already imports `utils`, but `utils` would now import `catalog`... 

So either we leave it as is, or we move `generate_id` to `utils`. There might also be another solution which I'm not thinking about.

Reply:
I think moving `generate_id` to utils make sense because we use it in many places.

Reply:
Actually I don't know if this is any better, since `generate_id` needs `ID_COLUMNS`, which is in `catalog`...

Reply:
What if you do `import catalog`  instead of `from .catalog` ?

https://stackoverflow.com/q/5748946/17083204

Reply:
Tu peux importer `catalog` à l'intérieur de la fonction, au lieu du top du fichier?

Reply:
J'avais en mémoire que `pre-commit` allait le redéplacer en haut, mais ça semble avoir été accepté.